### ISSUE TITLE
Enha: additional seeds and crops

### ISSUE THREAD

**Issue by HubertReX** [sloukit/pydew-valley-uzh#248]
### What would you like to see changed in one sentence?

all seeds and crops needed for first study 3a (v1, v2, v3)

### Elaborate on your suggestion above as needed.

The game currently supports 2 types of seeds and crops:
- tomatos,
- corn

We also need:
- beetroot, 
- carrot, 
- eggplant, 
- pumpkin, 
- parsnip

Sprites for all the above types of crops are available in `images/objects/items`. It's just a matter of extending all enums and making sure they are properly displayed in the Inventory menu.

### Which aspect(s) of the project is this enhancement affecting?

game-ui, game-gameplay

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by HubertReX** [comment]
@sloukit what about those existing crops:
- WOOD,
- APPLE,
- BLACKBERRY,
- BLUEBERRY,
- RASPBERRY,
- ORANGE,
- PEACH,
- PEAR

They are not mentioned in [excel sheet](https://docs.google.com/spreadsheets/d/1LEonXCrbTIiR7WGMviu3pL1GqCIjDiRld8wPUiq6j3Q/edit?usp=sharing) - do we want to keep them or hide in study 3a?

**Comment by sloukit** [comment]
> @sloukit what about those existing crops:
> 
> * WOOD,
> * APPLE,
> * BLACKBERRY,
> * BLUEBERRY,
> * RASPBERRY,
> * ORANGE,
> * PEACH,
> * PEAR
> 
> They are not mentioned in [excel sheet](https://docs.google.com/spreadsheets/d/1LEonXCrbTIiR7WGMviu3pL1GqCIjDiRld8wPUiq6j3Q/edit?usp=sharing) - do we want to keep them or hide in study 3a?

You can hide the crops that are not in the excel sheet for study 3a!

**Comment by HubertReX** [comment]
@sloukit  
crops that need to be added need to have their own growth speed:
```json
{
    "corn": 1,
    "tomato": 0.7,

    "beetroot": 0.7,
    "carrot": 0.7,
    "eggplant": 0.7,
    "pumpkin": 0.7,
    "parsnip": 0.7,
}
```

and value at the market:
```
20,  # CORN
40,  # TOMATO

20,  # BEETROOT
20,  # CARROT
20,  # EGGPLANT
20,  # PUMPKIN
20,  # PARSNIP

4,  # CORN_SEED
5,  # TOMATO_SEED

6,  # BEETROOT_SEED
6,  # CARROT_SEED
6,  # EGGPLANT_SEED
6,  # PUMPKIN_SEED
6,  # PARSNIP_SEED

```
please review the values.

**Comment by HubertReX** [comment]
> > @sloukit what about those existing crops:
> > 
> > * WOOD,
> > * APPLE,
> > * BLACKBERRY,
> > * BLUEBERRY,
> > * RASPBERRY,
> > * ORANGE,
> > * PEACH,
> > * PEAR
> > 
> > They are not mentioned in [excel sheet](https://docs.google.com/spreadsheets/d/1LEonXCrbTIiR7WGMviu3pL1GqCIjDiRld8wPUiq6j3Q/edit?usp=sharing) - do we want to keep them or hide in study 3a?
> 
> You can hide the crops that are not in the excel sheet for study 3a!

After playing the game for a bit longer, I realized that those crops actually play some role. They are collectibles that can be sold on the market, and they can be gathered in the forest. Even the tutorial is encouraging to do so. So now, I'd rather suggest keeping them always available. There are no seeds for those, so the player can't plant them and it's not interfering with the idea of making one new crop available with each game level.

**Comment by HubertReX** [comment]
There is one last concern related to crops and the market. In the market we can see the number of items we possess, we can buy seeds and sell crops. However player can't see the price and amount of money he possesses although it is calculated. At some point, we can run out of money and won't be able to buy goods, but without any information about why. Is this supposed to be like this or should I create a bug for it?

**Comment by sloukit** [comment]
> @sloukit crops that need to be added need to have their own growth speed:
> 
> ```json
> {
>     "corn": 1,
>     "tomato": 0.7,
> 
>     "beetroot": 0.7,
>     "carrot": 0.7,
>     "eggplant": 0.7,
>     "pumpkin": 0.7,
>     "parsnip": 0.7,
> }
> ```
> 
> and value at the market:
> 
> ```
> 20,  # CORN
> 40,  # TOMATO
> 
> 20,  # BEETROOT
> 20,  # CARROT
> 20,  # EGGPLANT
> 20,  # PUMPKIN
> 20,  # PARSNIP
> 
> 4,  # CORN_SEED
> 5,  # TOMATO_SEED
> 
> 6,  # BEETROOT_SEED
> 6,  # CARROT_SEED
> 6,  # EGGPLANT_SEED
> 6,  # PUMPKIN_SEED
> 6,  # PARSNIP_SEED
> ```
> 
> please review the values.

That seems perfectly fine to me, thank you!

**Comment by sloukit** [comment]
> > > @sloukit what about those existing crops:
> > > 
> > > * WOOD,
> > > * APPLE,
> > > * BLACKBERRY,
> > > * BLUEBERRY,
> > > * RASPBERRY,
> > > * ORANGE,
> > > * PEACH,
> > > * PEAR
> > > 
> > > They are not mentioned in [excel sheet](https://docs.google.com/spreadsheets/d/1LEonXCrbTIiR7WGMviu3pL1GqCIjDiRld8wPUiq6j3Q/edit?usp=sharing) - do we want to keep them or hide in study 3a?
> > 
> > 
> > You can hide the crops that are not in the excel sheet for study 3a!
> 
> After playing the game for a bit longer, I realized that those crops actually play some role. They are collectibles that can be sold on the market, and they can be gathered in the forest. Even the tutorial is encouraging to do so. So now, I'd rather suggest keeping them always available. There are no seeds for those, so the player can't plant them and it's not interfering with the idea of making one new crop available with each game level.

Yes, you're right. I'd keep those fruits in the game too, at all times. You could add them to the excel sheet, just so we can keep track of what is included.

**Comment by sloukit** [comment]
> There is one last concern related to crops and the market. In the market we can see the number of items we possess, we can buy seeds and sell crops. However player can't see the price and amount of money he possesses although it is calculated. At some point, we can run out of money and won't be able to buy goods, but without any information about why. Is this supposed to be like this or should I create a bug for it?
![Bildschirmfoto 2024-12-22 um 10 44 36](https://github.com/user-attachments/assets/b240e9b8-49f8-4454-a9a3-b909c507c235)


Thanks for pointing this out, I hadn't noticed - at one point the amount of money was definitely displayed in the market (see screenshot). So it must have been removed at some point. I'll try to find out where this happened and let you know.
![Bildschirmfoto 2024-12-22 um 10 44 36](https://github.com/user-attachments/assets/43c47a4f-b317-4911-8b4d-4442c19f2f7b)

**Comment by HubertReX** [comment]
> > > > @sloukit what about those existing crops:
> > > > 
> > > > * WOOD,
> > > > * APPLE,
> > > > * BLACKBERRY,
> > > > * BLUEBERRY,
> > > > * RASPBERRY,
> > > > * ORANGE,
> > > > * PEACH,
> > > > * PEAR
> > > > 
> > > > They are not mentioned in [excel sheet](https://docs.google.com/spreadsheets/d/1LEonXCrbTIiR7WGMviu3pL1GqCIjDiRld8wPUiq6j3Q/edit?usp=sharing) - do we want to keep them or hide in study 3a?
> > > 
> > > 
> > > You can hide the crops that are not in the excel sheet for study 3a!
> > 
> > 
> > After playing the game for a bit longer, I realized that those crops actually play some role. They are collectibles that can be sold on the market, and they can be gathered in the forest. Even the tutorial is encouraging to do so. So now, I'd rather suggest keeping them always available. There are no seeds for those, so the player can't plant them and it's not interfering with the idea of making one new crop available with each game level.
> 
> Yes, you're right. I'd keep those fruits in the game too, at all times. You could add them to the excel sheet, just so we can keep track of what is included.

ok, but please don't add them to the Excel, I've already implemented them as default list of crops

**Comment by HubertReX** [comment]
> > There is one last concern related to crops and the market. In the market we can see the number of items we possess, we can buy seeds and sell crops. However player can't see the price and amount of money he possesses although it is calculated. At some point, we can run out of money and won't be able to buy goods, but without any information about why. Is this supposed to be like this or should I create a bug for it?
> 
> Thanks for pointing this out, I hadn't noticed - at one point the amount of money was definitely displayed in the market (see screenshot). So it must have been removed at some point. I'll try to find out where this happened and let you know. ![Bildschirmfoto 2024-12-22 um 10 44 36](https://private-user-images.githubusercontent.com/120415680/397976549-43c47a4f-b317-4911-8b4d-4442c19f2f7b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjEwNzYsIm5iZiI6MTczNDg2MDc3NiwicGF0aCI6Ii8xMjA0MTU2ODAvMzk3OTc2NTQ5LTQzYzQ3YTRmLWIzMTctNDkxMS04YjRkLTQ0NDJjMTlmMmY3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyMlQwOTQ2MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNWY1YWNlZmIyM2QzODU4YjFkNTljYTE1M2NmMzc2NWYzZDE0MjcyNWViMzY5YTQ2YTdkMDkzZTY1ZjMwOTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FiaZLqYlX_qVYzorizHWJh544o38C6CEKB0U91x_qbk)

I see, so now, the list of crops is so long that the amount of money is below the bottom of the screen. I'll fix it. Unfortunately, the Inventory menu needs more refactoring, there is no scrolling implemented to handle more crops than can fit on the screen.

**Comment by sloukit** [comment]
> > > There is one last concern related to crops and the market. In the market we can see the number of items we possess, we can buy seeds and sell crops. However player can't see the price and amount of money he possesses although it is calculated. At some point, we can run out of money and won't be able to buy goods, but without any information about why. Is this supposed to be like this or should I create a bug for it?
> > 
> > 
> > Thanks for pointing this out, I hadn't noticed - at one point the amount of money was definitely displayed in the market (see screenshot). So it must have been removed at some point. I'll try to find out where this happened and let you know. ![Bildschirmfoto 2024-12-22 um 10 44 36](https://private-user-images.githubusercontent.com/120415680/397976549-43c47a4f-b317-4911-8b4d-4442c19f2f7b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjEwNzYsIm5iZiI6MTczNDg2MDc3NiwicGF0aCI6Ii8xMjA0MTU2ODAvMzk3OTc2NTQ5LTQzYzQ3YTRmLWIzMTctNDkxMS04YjRkLTQ0NDJjMTlmMmY3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyMlQwOTQ2MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNWY1YWNlZmIyM2QzODU4YjFkNTljYTE1M2NmMzc2NWYzZDE0MjcyNWViMzY5YTQ2YTdkMDkzZTY1ZjMwOTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FiaZLqYlX_qVYzorizHWJh544o38C6CEKB0U91x_qbk)
> 
> I see, so now, the list of crops is so long that the amount of money is below the bottom of the screen. I'll fix it. Unfortunately, the Inventory menu needs more refactoring, there is no scrolling implemented to handle more crops than can fit on the screen.

Okay, I also just noticed that the amount of money is still there, but just too far down. We already have a scrolling system implemented in this [PR](https://github.com/sloukit/pydew-valley-uzh/pull/162). It might save some time to just copy it.

**Comment by HubertReX** [comment]
> Okay, I also just noticed that the amount of money is still there, but just too far down. We already have a scrolling system implemented in this [PR](https://github.com/sloukit/pydew-valley-uzh/pull/162). It might save some time to just copy it.

Yes, I've seen it in other menus, but it seams that each menu is implemented a bit differently. Still, it's helpful to reuse this code 💪 Thanks for the tip!

**Comment by HubertReX** [comment]
this issue has been fixed in PR #249. 

Crops that are visible depend on the game level and round (level). 
In the shop menu (press `y` in debug mode), there are some additional features added:
- list scrolling
- menu title and column headers
- additional column with crop value
- crop icon on the left of crop name (as an alternative version, it can be moved inside the white bar, but then it needs to be scaled down by 2)
- action label changed from 'buy' / 'sell' to 'buy one' / 'sell one'
- footer label 'your money' added

after a review of the PR, this issue can be closed

**Comment by HubertReX** [comment]
PR #249 has been merged, it's fixed now
